src/libpmemset/pmemset.c

```diff
@@ -908,14 +952,32 @@ pmemset_remove_part_map_range_cb(struct pmemset *set,
 		return PMEMSET_E_PART_MAP_POSSIBLE_USE_AFTER_DROP;
 	}
 
+	int ret;
 	struct pmap_remove_range_arg *rarg =
 			(struct pmap_remove_range_arg *)arg;
 	size_t rm_addr = rarg->addr;
 	size_t rm_size = rarg->size;
 
+	struct pmem2_vm_reservation *pmem2_reserv = pmap->pmem2_reserv;
 	size_t pmap_addr = (size_t)pmap->desc.addr;
 	size_t pmap_size = pmap->desc.size;
-	struct pmem2_vm_reservation *pmem2_reserv = pmap->pmem2_reserv;
+
+	struct pmem2_map *first_p2map;
+	struct pmem2_map *last_p2map;
+	ret = pmemset_part_map_find(pmap, 0, 1, &first_p2map);
+	ASSERTeq(ret, 0);
+
+	ret = pmemset_part_map_find(pmap, pmap_size - 1, 1, &last_p2map);
+	ASSERTeq(ret, 0);
+
+	size_t first_p2map_end_addr = (size_t)pmem2_map_get_address(first_p2map)
+			+ pmem2_map_get_size(first_p2map);
+	size_t last_p2map_addr = (size_t)pmem2_map_get_address(last_p2map);
+
+	/* range covers pmem2 maps at the start and the end of part map */
+	if (rm_addr < first_p2map_end_addr &&
+			rm_addr + rm_size > last_p2map_addr)
+		return pmemset_remove_part_map_no_lock(set, &pmap);
```

Comment:
Is this fixing some unrelated bug or is it just a refactor? The old code had to remove the part anyway, right?